src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+   */
+  style?: React.CSSProperties;
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  backgroundColor = colors.silver.light,
+  children,
+  disabled = false,
+  disabledBackgroundColor = colors.silver.dark,
+  disabledTextColor = colors.grey.light,
+  fab = false,
+  forceActiveState = false,
+  forceFocusState = false,
+  forceHoverState = false,
+  hidden = false,
+  hiddenActiveBackgroundColor = colors.silver.dark,
+  hiddenFocusedColor = colors.blue.base,
+  hoverBackgroundColor = colors.silver.base,
+  icon,
+  size = "default",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  if (forceActiveState && forceHoverState) {
+    throw new TypeError(
+      "Do not force multiple properties at once, you will get indeterminiate behavior"
+    );
+  }
+
+  // We're storing this because it can be used for the `:active` style or when
+  // `forceActiveState` is used in Storybook
+  const activeProperties = {
+    ...(hidden
+      ? { backgroundColor: hiddenActiveBackgroundColor }
+      : {
+          // The `box-shadow` property is copied directly from Zeplin
+          boxShadow:
+            "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
+        }),
+    outline: 0
+  };
+
+  // We're storing this because it can be used for the `:hover` style or when
+  // `forceHoverState` is used in Storybook
+  const hoverProperties = {
+    backgroundColor: hoverBackgroundColor,
+    cursor: "pointer",
+    ...(!hidden && {
+      // The `box-shadow` property is copied directly from Zeplin
+      boxShadow:
+        "0 5px 10px 0 rgba(18, 21, 26, 0.12), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+    })
+  };
+
+  // We're storing this because it can be used for the `:focus` style or when
+  // `forceFocusState` is used in Storybook
+  const focusProperties = {
+    // The `box-shadow` property is copied directly from Zeplin
+    boxShadow:
+      "0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 0 0 2px #bbdbff, inset 0 0 0 1px #2075d6, inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)",
+    ...(hidden && { color: `${hiddenFocusedColor} !important` })
+  };
+
+  return (
+    <button
+      {...otherProps}
+      css={{
+        appearance: "button",
+        backgroundColor: hidden
```

Comment:
If we define disabled styles under the `[disabled]` selector, this ternary could be simplified:

```js
{
  backgroundColor: hidden ? 'transparent' : backgroundColor,
  '[disabled]': {
    backgroundColor: disabledBackgroundColor
  }
}
```